### PR TITLE
Release `kedro-datasets 1.0.0`

### PR DESCRIPTION
**Pull request by ankatiyar**
Signed-off-by: Ankita Katiyar <ankitakatiyar2401@gmail.com>

## Description

Release `kedro-datasets 1.0.0`, now that Kedro 0.18.4 is out! 

## Development notes

- update version to `1.0.0`
- pin `kedro 0.18.4` to requirements
- release notes
- `readme.md` for `kedro-plugins` to add `kedro-datasets` to the list
- `readme.md` for `kedro-datasets` 

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-kedro
+kedro==0.18.4
```
Comment (ankatiyar) [inline kedro-datasets/requirements.txt:1]: not sure if this is required?

Comment (merelcht) [reply kedro-datasets/requirements.txt:1]: I think it can be `kedro~=0.18.4` as @noklam wrote in https://github.com/kedro-org/kedro/issues/2004

Comment (ankatiyar) [reply kedro-datasets/requirements.txt]: done!

**Review by SajidAlamQB — APPROVED** [review]

**Comment by merelcht** [comment]
It would be nice to add an entry to the release notes describing what `kedro-datasets` is.

**Comment by ankatiyar** [comment]
> It would be nice to add an entry to the release notes describing what `kedro-datasets` is.

I was also thinking of adding `kedro-datasets` to the [`kedro-plugins` README file] (https://github.com/kedro-org/kedro-plugins/blob/main/README.md) in this PR, what do you think?

**Comment by merelcht** [comment]
> > It would be nice to add an entry to the release notes describing what `kedro-datasets` is.
> 
> I was also thinking of adding `kedro-datasets` to the [`kedro-plugins` README file](https://github.com/kedro-org/kedro-plugins/blob/main/README.md) in this PR, what do you think?

Yes that would be great! 

I think in terms of release notes you can write something like:

# Release 1.0.0

First official release of Kedro-Datasets.

Datasets are Kedro’s way of dealing with input and output in a data and machine-learning pipeline. [Kedro supports numerous datasets](https://kedro.readthedocs.io/en/stable/kedro.extras.datasets.html) out of the box to allow you to process different data formats including Pandas, Plotly, Spark and more.

**Review by merelcht — APPROVED** [review]
⭐ ⭐ ⭐

**Pull request merged**